authors/apps/article_likes/views.py

```diff
@@ -0,0 +1,126 @@
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from authors.apps.articles.models import Article
+from authors.apps.articles.serializers import ArticleSerializer
+from .models import ArticleLike
+from .serializers import ArticleLikeSerializer
+
+class ArticleLikingViews(APIView):
```

Comment:
@AtamaZack, I like the way you have implemented the like functionality, however, I noticed some duplication. I noticed that the two classes ('ArticleLikingViews' and 'ArticleDislikingViews') have similar pieces of code. I believe we can abstract the code further to remove the duplication.
I have tried to work out a way on how to remove the duplication while maintaining the amazing functionality that you came up with.
How do you see if we parameterize the like and dislike in the URL and then use the parameterized string to determine whether the user wants to like or dislike or revoke their like or dislike?
Then, we can also use one class to implement the functionality but first, we need to analyze the similar pieces of code in classes 'ArticleLikingViews' and 'ArticleDislikingViews' and the form one class with parameterized sections for the URL parameters. 
URL - `path("<like_status>", ArticleLikeDislikeViews.as_view())`

Reply:
Thanks for the review @kalsmic, onto it.